src/mark.js

```diff
@@ -658,6 +732,88 @@ class Mark { // eslint-disable-line no-unused-vars
     }
 
     /**
+     * Callback for each wrapped element
+     * @callback Mark~wrapRangeFromIndexCallback
+     * @param {HTMLElement} element - The marked DOM element
+     */
+    /**
+     * Filter callback before each wrapping
+     * @callback Mark~wrapRangeFromIndexFilterCallback
+     * @param {object} range - object containing the range start, and end or
+     * length
+     * @param {string} match - string extracted from the matching range
+     * @param {HTMLElement} node - The text node which includes the range
+     * @param {number} counter - A counter indicating the number of all marks
+     */
+    /**
+     * Callback on end
+     * @callback Mark~wrapRangeFromIndexEndCallback
+     */
+    /**
+     * Wraps the indicated ranges across all HTML elements in all contexts
+     * @param {Mark~setOfRanges} ranges
+     * @param {Mark~wrapRangeFromIndexFilterCallback} filterCb
+     * @param {Mark~wrapRangeFromIndexEachCallback} eachCb
+     * @param {Mark~wrapRangeFromIndexEndCallback} endCb
+     * @access protected
+     */
+    wrapRangeFromIndex(ranges, filterCb, eachCb, endCb) {
```

Comment:
> not adding attributes inside this function

Well, we can use the `each` callback to accomplish the same thing, so maybe completely removing it would be better.